Source/UI/SelectableText.swift

```diff
@@ -0,0 +1,75 @@
+//
+//  SelectableText.swift
+//  Planetary
+//
+//  Created by Martin Dutra on 20/9/22.
+//  Copyright © 2022 Verse Communications Inc. All rights reserved.
+//
+
+import Foundation
+import SwiftUI
+import UIKit
+
+/// A ViewRepresentable that wraps a UILabel meant to be used in place of Text when selection word by word is desired.
+///
+/// SwiftUI's Text cannot be configured to be selected in a word by word basis, just the whole text, wrapping up a
+/// UILabel achieves this. Also, this is configured to use a monospaced font as the intended use at the moment is
+/// when showing a Source Message.
+struct SelectableText: UIViewRepresentable {
+
+    var text: String
+
+    init(text: String) {
+        self.text = text
+    }
+
+    private var font = UIFont.monospacedSystemFont(ofSize: 12, weight: .regular)
```

Comment:
Can we use the preferred font API here to support Dynamic Type? Like this:
```suggestion
    private var font = UIFont.preferredFont(forTextStyle: .body)
```

Reply:
I want a monospaced font for the message source, I didn't find way to get that using preferredFont and there isn't a text style for code. I'd say that this is part of a broader issue of how to make SelectableText more generic so that we can use it from other places as well, ones with a regular text, others with monospaced, and it would be rather fixed once we want to reuse it.

Reply:
Ah, that makes sense. I think we can define our own Dynamic Type fonts and we could do one that's monospaced. But I don't think we need to figure it out right now, unless you just want to.